### PR TITLE
Hardfork testnet to force upgrade

### PR DESCRIPTION
**Pull request by evan82**
This should help us to force the network to upgrade and remove all of the old non-compliant nodes from our test group.e Our masternode inputs will still be valid, so it's easy to update. 

Fork initiates at block 80k and is implemented by adding one required KGW block.

### CODE REVIEW

File: src/pow.cpp [hunk 1]
```diff
@@ -147,7 +147,8 @@ unsigned int GetNextWorkRequired(const CBlockIndex* pindexLast, const CBlockHead
         else retarget = DIFF_BTC;
     // testnet -- we want a lot of coins in existance early on 
     } else {
-        if (pindexLast->nHeight + 1 >= 3000) retarget = DIFF_DGW;
+        if (pindexLast->nHeight + 1 >= 80000) retarget = DIFF_KGW;
```
Comment (tgflynn) [inline src/pow.cpp]: It looks like this switches to KGW after 80K.  Is that what we want ?  It doesn't seem to match the commit message.

**Comment by UdjinM6** [comment]
In general, I'm not sure we need this. Wouldn't proto bump + spork be enough?

**Comment by schinzelh** [comment]
I agree, i don't see why we need a hard fork here - will we hard fork mainnet on 12.1 rollout as well? I assume not :-)

nACK

**Comment by tgflynn** [comment]
@schinzelh Mainnet will have to fork at some point because 12.0 nodes won't recognize new superblock payments and vice versa.  We could cause a testnet fork simply be enabling superblocks but I think that if we do that now with some nodes running watchdog and some not we will probably end up with multiple forks because of disagreements over which MN's are valid or not.  Changing a block parameter may be a cleaner way to do it.  

That said I'm also not convinced this is necessary.  Can't we just make an announcement to ask more people to upgrade like we did before ?  In any case I think if we do this hard fork it should target a future block, not a past one.

**Comment by schinzelh** [comment]
@tgflynn you are right, my bad: Mainnet obviously needs to "spork" for the new superblocks.

Anyway: the update of testnet should be "enforced" with the same tools as we will do on mainnet - and i doubt we will change to KGW on mainnet...

Protobump and spork it is for me...

**Comment by evan82** [comment]
In the past I've found it helpful to have a separate enclave for testing each of the major changes. It ensures old protocol versions are detached from our test is all.

**Pull request closed without merging**